// Copyright 2022 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package url_param_filter;

message FilterParameter {
  optional string name = 1;  // required, eg. the "foo" in "?foo=bar"
}

message FilterClassification {
  enum SiteRole {
    SITE_ROLE_UNKNOWN = 0;
    SOURCE = 1;
    DESTINATION = 2;
  }
  enum UseCase {
    USE_CASE_UNKNOWN = 0;
    CROSS_OTR = 1;
    CROSS_SITE_NO_3PC = 2;
  }
  enum SiteMatchType {
    MATCH_TYPE_UNKNOWN =
        0;  // Current behavior: default to EXACT_ETLD_PLUS_ONE.
    EXACT_ETLD_PLUS_ONE = 1;  // Exact match on eTLD+1.
    ETLD_WILDCARD = 2;        // Match on any eTLD, e.g. if site is "foo", match
                              // "foo.com" and "foo.io".
  }
  optional string site = 1;         // eTLD+1, required
  optional SiteRole site_role = 2;  // required
  optional SiteMatchType site_match_type = 6;
  repeated FilterParameter parameters = 3;
  repeated UseCase use_cases = 4 [packed = true];
  repeated string experiment_tags = 5;
}

message FilterClassifications {
  repeated FilterClassification classifications = 1;
}
